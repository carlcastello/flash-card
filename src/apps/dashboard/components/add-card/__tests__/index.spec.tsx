import React from "react";
import { shallow } from "enzyme";

import { AddCard } from "../index";


describe("Add Card", () => {

  let wrapper: any;
  let instance: any;
  let onClick: any;

  beforeEach(() => {
    onClick = jest.fn();
    wrapper = shallow(
      <AddCard
        onClick={onClick}
        classes={{
          paperContainer: 'paper-container',
          boxContainer: 'box-container'
        }}/>
    );
    instance = wrapper.instance();
  });

  describe("The Add Card", () => {
    it("renders", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("State after onMouseHoverAction", () => {
    it("Changes the state if hasHoverEffect", () => {
      expect(wrapper.state().paperElevation).toEqual(3);
      wrapper.setProps({hasHoverEffect: false});
      wrapper.update();
      instance.onMouseHoverAction(6)();
      expect(wrapper.state().paperElevation).toEqual(3);
    });

    it("Changes the state if hasHoverEffect", () => {
      expect(wrapper.state().paperElevation).toEqual(3);
      wrapper.setProps({hasHoverEffect: true});
      wrapper.update();
      instance.onMouseHoverAction(6)();
      expect(wrapper.state().paperElevation).toEqual(6);
    });
  });
});