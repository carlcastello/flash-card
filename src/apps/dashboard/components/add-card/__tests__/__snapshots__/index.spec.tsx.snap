// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Card The Add Card renders 1`] = `ShallowWrapper {}`;
